tasks/run-collectors/run-collectors.yaml

```diff
@@ -0,0 +1,99 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: run-collectors
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to run collectors defined in the ReleasePlan and ReleasePlanAdmission
+  params:
+    - name: releasePlanAdmissionPath
+      type: string
+      description: Path to the JSON string of the ReleasePlanAdmission in the data workspace
+    - name: releasePlanPath
+      type: string
+      description: Path to the JSON string of the ReleasePlan in the data workspace
+    - name: resultsDir
+      type: string
+      description: The relative path in the workspace to save the collector results to
+  workspaces:
+    - name: data
+      description: Workspace where the CRs are stored
+  steps:
+    - name: run-collectors
+      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      script: |
+        #!/usr/bin/env bash
+        set -xeo pipefail
+
+        DEFAULT_TIMEOUT=900
+        COLLECTORS_REPOSITORY="https://github.com/SOME_ORG/SOME_REPO" # TODO - update this once repo is created
+        RELEASEPLAN_FILE="$(workspaces.data.path)/$(params.releasePlanPath)"
+        RELEASEPLANADMISSION_FILE="$(workspaces.data.path)/$(params.releasePlanAdmissionPath)"
+        RESULTS_DIR="$(workspaces.data.path)/$(params.resultsDir)"
+        CONCURRENT_LIMIT=4
+
+        execute_collector() { # Expect argument is [collector json, type of collector]
+            collector="$1"
+            name="$(jq -r '.name' <<< "$collector")"
+            type="$(jq -r '.type' <<< "$collector")"
+            TIMEOUT="$(jq -r --arg default "$DEFAULT_TIMEOUT" '.timeout // $default' <<< "$collector")"
+            ARGS=()
+            NUM_ARGS="$(jq '.params | length' <<< "$collector")"
+            for ((j = 0; j < NUM_ARGS; j++)) ; do
+                param="$(jq -c --argjson j "$j" '.params[$j]' <<< "$collector")"
+                arg_name="$(jq -r '.name' <<< "$param")"
+                arg_value="$(jq -r '.value' <<< "$param")"
+                ARGS=("${ARGS[@]}" --"${arg_name}" "${arg_value}")
+            done
+            # Execute collector
+            timeout "$TIMEOUT" python3 "$type" "${ARGS[@]}" | tee "$RESULTS_DIR"/"$2"-"$name".json
```

Comment:
I guess we can change this later. We're probably not sure exactly what it will look like yet. But I think it would be a good idea to have a parameter for the output file in the python scripts. And it could log some stuff along the way (and of course an error if it fails) and we could save that to a file and print all the logs after the parallel runs finish - like we do here: https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/push-rpm-data-to-pyxis/push-rpm-data-to-pyxis.yaml#L171

Reply:
Yeah, I don't disagree. I think that is one of maybe multiple changes to this task once we have one actual collector implemented

Reply:
ok, cool, resolving for now.